.github/workflows/main.yml

```diff
@@ -14,7 +14,41 @@ jobs:
     # This action builds the container and executes the test suite inside it.
     - uses: ./.github/actions/test
       with:
-        target: test-full
+        target: test-ci
+
+    - name: Ensure testing did not change sources
+      run: git diff --exit-code
+
+  smoke_test_benchmarks:
+    name: Smoke-test benchmarks
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+      with:
+        submodules: 'recursive'
+
+    # Testing uses the development environment Docker container.
+    # This action builds the container and executes the test suite inside it.
+    - uses: ./.github/actions/test
+      with:
+        target: test-benchmarks
+
+    - name: Ensure testing did not change sources
+      run: git diff --exit-code
+
+  smoke_test_fuzz:
+    name: Smoke-test fuzz
+    runs-on: ubuntu-latest
```

Comment:
This eventually gets run in our `bionic` container, right?

Reply:
Yes. The `uses: ./.github/actions/test` step builds (without a cache, grr) and runs the container. The `runs-on` is just the host of the docker environment, afaict.